modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCartButton.js

```diff
@@ -38,10 +42,78 @@ function AddToCartButton({
 	onClick,
 	onError,
 	settings,
+	showOrderTypeModal,
+	showOrderTypeModalURL,
 }) {
 	const [cartAtomState, setCartAtomState] = useLiferayState(cartAtom);
 	const [isTriggeringCartUpdate, setIsTriggeringCartUpdate] = useState(false);
 	const isMounted = useIsMounted();
+	const [event, setEvent] = useState(null);
+
+	const addToCartNotAllowed =
+		cpInstances[0].quantity > cpInstances[0].stockQuantity;
```

Comment:
This was added to prevent the modal to open if the user add a higher quantity that is available in stock. If I remember correctly, the `AddToCart` button stays active and only after the click, a toast message is displayed.